### PR TITLE
Improve readme wasm32-unknown-unknown nightly section

### PR DESCRIPTION
**Pull request by rukai**
I made this section of the readme clearer based on your comment here https://github.com/Amanieu/parking_lot/issues/310#issuecomment-1004309603

I figured the primitives having no effect wasnt worth mentioning because wasm32-unknown-unknown is single threaded anyway.

Happy to reword or change anything I wrote.

### CODE REVIEW

**Comment by rukai** [comment]
Oh I thought `-C target-features=+atomics` was the nightly feature that caused the dependency on nightly.
If that is available on stable then what is nightly required for?

**Comment by bjorn3** [comment]
The standard library must be compiled with `-C target-features=+atomics` too, which requires the nightly only `-Zbuild-std` cargo flag.

**Comment by rukai** [comment]
Ah I see, alright, I updated the PR.

File: README.md [hunk 1]
```diff
@@ -94,8 +94,13 @@ There are a few restrictions when using this library on stable Rust:
 - You will have to use the `const_*` functions (e.g. `const_mutex(val)`) to
   statically initialize the locking primitives. Using e.g. `Mutex::new(val)`
   does not work on stable Rust yet.
-- The `wasm32-unknown-unknown` target is only supported on nightly and requires
-  `-C target-feature=+atomics` in `RUSTFLAGS`.
+- The `wasm32-unknown-unknown` target is only fully supported on nightly with
+  `-C target-feature=+atomics -Z build-std` in `RUSTFLAGS`.
```
Comment (bjorn3) [inline README.md:98]: `-Zbuild-std` is an argument to be passed directly to cargo or enabled in `.cargo/config.toml`. `-Ctarget-feature=+atomics` is an argument of rustc passed using the `RUSTFLAGS` env var.

Comment (rukai) [reply README.md]: ah, thanks, I split them up accordingly.

File: README.md [hunk 2]
```diff
@@ -94,8 +94,13 @@ There are a few restrictions when using this library on stable Rust:
...
   does not work on stable Rust yet.
-- The `wasm32-unknown-unknown` target is only supported on nightly and requires
-  `-C target-feature=+atomics` in `RUSTFLAGS`.
+- The `wasm32-unknown-unknown` target is only fully supported on nightly with
+  `-C target-feature=+atomics` in `RUSTFLAGS` and `-Z build-std` passed to cargo.
+  parking_lot will work mostly fine on stable, the only difference is it will
+  panic instead of block forever if you hit a deadlock.
+  Just make sure not to enable `-C target-feature=+atomics` on stable as that
+  will allow wasm to run with multiple threads which will completely break
+  parking_lots concurrency guarantees.
```
Comment (Amanieu) [inline README.md:103]:
```suggestion
  parking_lot's concurrency guarantees.
```

Comment (rukai) [reply README.md]: thanks, fixed

**Comment by Amanieu** [comment]
Thanks!

**Pull request merged**